[Bug]: Unit Test "test-install (windows|linux|macos-latest, 3.12)" occasionally fails

### Describe the bug

CI job failed during the installation test because the memmachine-server never responded at http://127.0.0.1:8080/api/v2/projects/list. The workflow spent over 5 minutes retrying, but received HTTP 000 (empty response) every time—meaning the server did not start, crashed, or could not bind to the required port. All earlier configuration steps seemed to pass, and the problem occurred across OSes.

Here is a screenshot

<img width="1998" height="4605" alt="Image" src="https://github.com/user-attachments/assets/0ebd9445-adab-4810-8cbc-86ef9398fd20" />

### Steps to reproduce

Recent Pull Requests have been failing and exhibit the same issues in the Installation Unit tests on Linux, Windows, and macOS.

eg: https://github.com/MemMachine/MemMachine/actions/runs/20828093367/job/59834820740?pr=925

### Expected behavior

memmachine-server should start and respond at the required endpoint within the allowed retries in CI.

### Environment

- GitHub Actions

### Additional context

Here is the MemMachine log file showing the cause of why MemMachine fails to start. 

[server-out.log](https://github.com/user-attachments/files/24509332/server-out.log)

```
INFO:memmachine.common.configuration:Loading configuration from 'C:\Users\runneradmin\.config\memmachine\cfg.yml'
INFO:memmachine.common.configuration.log_conf:applying log configuration: level=INFO, format=%(asctime)s [%(levelname)s] %(name)s - %(message)s, path=MemMachine.log
2026-01-08 19:00:31,205 [INFO] memmachine.server.api_v2.mcp - Configuration file 'C:\Users\runneradmin\.config\memmachine\cfg.yml' loaded.
2026-01-08 19:00:31,205 [INFO] memmachine.server.app - Starting server with 1 worker
C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\websockets\legacy\__init__.py:6: DeprecationWarning: websockets.legacy is deprecated; see https://websockets.readthedocs.io/en/stable/howto/upgrade.html for upgrade instructions
  warnings.warn(  # deprecated in 14.0 - 2024-11-09
C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\uvicorn\protocols\websockets\websockets_impl.py:17: DeprecationWarning: websockets.server.WebSocketServerProtocol is deprecated
  from websockets.server import WebSocketServerProtocol
INFO:     Started server process [5072]
INFO:     Waiting for application startup.
2026-01-08 19:00:31,238 [INFO] memmachine.common.configuration - Loading configuration from 'C:\Users\runneradmin\.config\memmachine\cfg.yml'
2026-01-08 19:00:31,246 [INFO] memmachine.common.configuration.log_conf - applying log configuration: level=INFO, format=%(asctime)s [%(levelname)s] %(name)s - %(message)s, path=MemMachine.log
2026-01-08 19:00:31,246 [INFO] memmachine.server.api_v2.mcp - Configuration file 'C:\Users\runneradmin\.config\memmachine\cfg.yml' loaded.
2026-01-08 19:00:31,246 [INFO] memmachine.common.configuration.log_conf - applying log configuration: level=INFO, format=%(asctime)s [%(levelname)s] %(name)s - %(message)s, path=MemMachine.log
2026-01-08 19:00:31,359 [INFO] memmachine.common.resource_manager.database_manager - Validating Neo4j driver 'neo4j_db'
2026-01-08 19:00:34,861 [INFO] memmachine.common.resource_manager.database_manager - Neo4j driver 'neo4j_db' validated successfully
2026-01-08 19:00:36,023 [WARNING] neo4j.notifications - Received notification from DBMS server: <GqlStatusObject gql_status='01N52', status_description='warn: property key does not exist. The property `dimensions` does not exist in database `neo4j`. Verify that the spelling is correct.', position=<SummaryInputPosition line=1, column=53, offset=52>, raw_classification='UNRECOGNIZED', classification=<NotificationClassification.UNRECOGNIZED: 'UNRECOGNIZED'>, raw_severity='WARNING', severity=<NotificationSeverity.WARNING: 'WARNING'>, diagnostic_record={'_classification': 'UNRECOGNIZED', '_severity': 'WARNING', '_position': {'offset': 52, 'line': 1, 'column': 53}, 'OPERATION': '', 'OPERATION_CODE': '0', 'CURRENT_SCHEMA': '/'}> for query: 'MATCH (s:SetEmbedding) RETURN s.set_id AS set_id, s.dimensions AS dims'
2026-01-08 19:00:36,412 [INFO] memmachine.common.resource_manager.embedder_manager - Validating embedder 'my_embedder' is working.
2026-01-08 19:00:36,918 [INFO] httpx - HTTP Request: POST http://localhost:11434/v1/embeddings "HTTP/1.1 404 Not Found"
2026-01-08 19:00:36,919 [ERROR] memmachine.common.embedder.openai_embedder - [call uuid: b8b48ef5-31c8-4263-961a-285e4655a143] Giving up creating embeddings for cluster number 0 after failed attempt 1 due to non-retryable NotFoundError
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\embedder\openai_embedder.py", line 236, in _embed_chunk_cluster
    await self._client.embeddings.create(
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\openai\resources\embeddings.py", line 251, in create
    return await self._post(
           ^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\openai\_base_client.py", line 1797, in post
    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\openai\_base_client.py", line 1597, in request
    raise self._make_status_error_from_response(err.response) from None
openai.NotFoundError: Error code: 404 - {'error': {'message': 'model "nomic-embed-text" not found, try pulling it first', 'type': 'api_error', 'param': None, 'code': None}}
ERROR:    Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\embedder\openai_embedder.py", line 236, in _embed_chunk_cluster
    await self._client.embeddings.create(
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\openai\resources\embeddings.py", line 251, in create
    return await self._post(
           ^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\openai\_base_client.py", line 1797, in post
    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\openai\_base_client.py", line 1597, in request
    raise self._make_status_error_from_response(err.response) from None
openai.NotFoundError: Error code: 404 - {'error': {'message': 'model "nomic-embed-text" not found, try pulling it first', 'type': 'api_error', 'param': None, 'code': None}}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\resource_manager\embedder_manager.py", line 64, in _validate_embedder
    _ = await embedder.search_embed(["a"])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\embedder\openai_embedder.py", line 126, in search_embed
    return await self._embed(queries, max_attempts)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\embedder\openai_embedder.py", line 177, in _embed
    clusters_chunk_embeddings = await asyncio.gather(
                                ^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\embedder\openai_embedder.py", line 300, in _embed_chunk_cluster
    raise ExternalServiceAPIError(error_message) from err
memmachine.common.data_types.ExternalServiceAPIError: [call uuid: b8b48ef5-31c8-4263-961a-285e4655a143] Giving up creating embeddings for cluster number 0 after failed attempt 1 due to non-retryable NotFoundError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\starlette\routing.py", line 694, in lifespan
    async with self.lifespan_context(app) as maybe_state:
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\fastapi\routing.py", line 153, in merged_lifespan
    async with original_context(app) as maybe_original_state:
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\server\api_v2\mcp.py", line 378, in mcp_http_lifespan
    async with global_memory_lifespan(), mcp_app.lifespan(application):
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\server\api_v2\mcp.py", line 344, in global_memory_lifespan
    await init_global_memory()
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\server\api_v2\mcp.py", line 354, in init_global_memory
    await mem_machine.start()
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\main\memmachine.py", line 120, in start
    semantic_service = await self._resources.get_semantic_service()
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\resource_manager\resource_manager.py", line 184, in get_semantic_service
    return await semantic_manager.get_semantic_service()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\resource_manager\semantic_manager.py", line 118, in get_semantic_service
    resource_retriever = await self.get_semantic_session_resource_manager()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\resource_manager\semantic_manager.py", line 67, in get_semantic_session_resource_manager
    default_embedder = await self._resource_manager.get_embedder(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\resource_manager\resource_manager.py", line 115, in get_embedder
    return await self._embedder_manager.get_embedder(name, validate=validate)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\resource_manager\embedder_manager.py", line 55, in get_embedder
    embedder = await self._build_embedder(name, validate=validate)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\resource_manager\embedder_manager.py", line 81, in _build_embedder
    await self._validate_embedder(name, ret)
  File "C:\hostedtoolcache\windows\Python\3.12.10\x64\Lib\site-packages\memmachine\common\resource_manager\embedder_manager.py", line 67, in _validate_embedder
    raise InvalidEmbedderError(f"embedder '{name}' is invalid. {e}") from e
memmachine.common.errors.InvalidEmbedderError: embedder 'my_embedder' is invalid. [call uuid: b8b48ef5-31c8-4263-961a-285e4655a143] Giving up creating embeddings for cluster number 0 after failed attempt 1 due to non-retryable NotFoundError

ERROR:    Application startup failed. Exiting.
```

--- a/.github/workflows/installation-test.yml
+++ b/.github/workflows/installation-test.yml
@@ -155,7 +155,7 @@ jobs:
         with:
           max_attempts: 5
           retry_wait_seconds: 10
-          timeout_seconds: 60
+          timeout_seconds: 300
           shell: bash
           command: |
             set -eo pipefail

